### ISSUE TITLE
EPIC: Steve should support per-resource schemas

### ISSUE THREAD

**Issue by richard-cox** [rancher/rancher#45827]
For details please see SURE-6164

Related Issues

`rancher` | blocking `dashboard` | SURE | Requirement
----|----|-----|-----
~https://github.com/rancher/rancher/issues/40712~ | ~https://github.com/rancher/dashboard/issues/7897~  | ~SURE-3183~ 
https://github.com/rancher/rancher/issues/34731 | https://github.com/rancher/dashboard/issues/3609 |  | Can signed in user create a project in a specific cluster
~https://github.com/rancher/rancher/issues/39378~ | | ~SURE-5416~
&nbsp; | https://github.com/rancher/dashboard/issues/5306 | | Can signed in user create a project or namespace (outside of a project) in a specific cluster
&nbsp;  | https://github.com/rancher/dashboard/issues/8594 | | Can signed in user manage project membership in a specific project
&nbsp; | https://github.com/rancher/dashboard/issues/4399 | | Can signed in user manage project membership in a specific project
https://github.com/rancher/rancher/issues/34731| https://github.com/rancher/dashboard/issues/5722 | | Can signed in user edit a project, or manage project membership
&nbsp; | https://github.com/rancher/dashboard/issues/6667 | | Can signed in user create a catalog.cattle.io app in a specific namespace
&nbsp; | https://github.com/rancher/dashboard/issues/8544 | SURE-6067 | Can signed in user create a secret in a namespace
&nbsp; | ~https://github.com/rancher/dashboard/issues/8606~ | | ~Can signed in user GET a resource that's supplied via a LIST request~
https://github.com/rancher/rancher/issues/48788 | https://github.com/rancher/dashboard/issues/12458 | SURE-8995 | Can signed in user manage project membership in a specific project

Issue of note https://github.com/rancher/rancher/issues/36799

[It would be nice if each of the above lines had a checkbox but github markdown doesn't support active task lists in tables (see https://github.com/github/task_list/issues/42 , closed as wontfix 2017-11-01 - EP]

**Comment by richard-cox** [comment]
From a UI perspective we need to understand if the user can do action x to resource y in context z. For example

- Add a user as a member of a project in a downstream cluster
- Remove a user as an owner from a cluster
- Remove a user's restricted-admin's role
- Edit a project (i don't think this is achievable via the resource's link.update value). Update: link's `update` url doesn't provide fine grain PUT/PATCH permissions) in a downstream cluster
- Add a configmap in a namespace in a downstream cluster
- Historically this was done via either a global schema applying to all resources, or the contents of a resource's actions/links collection. We would receive updates to both over socket if anything changed.

We need to work together to understand how this can be done in a way that scales, automatically updates and integrates with the UI well

**Comment by gaktive** [comment]
Internal reference: SURE-6164 for this work. SURE-8995 has picked up activity as of late and SURE-6656 rides in its wake.

**Comment by MKlimuszka** [comment]
https://github.com/rancher/rancher/issues/47393 is related to these issues, and *may* be tied to / fixed alongside https://github.com/rancher/rancher/issues/34731

**Comment by rohitsuse** [comment]
@crobby @richard-cox  This epic can be closed right ? since all the issues in the table are closed.

**Comment by richard-cox** [comment]
i've marked all the dashboard issues as unblocked.

Would be ok with this one closing
